ParametricObjectCreation/HoneycombSolid/HoneycombSolid.py

```diff
@@ -0,0 +1,235 @@
+							x_delta = centers_distance *  math.sin(edges_angle)
+							y_delta = centers_distance * math.cos(edges_angle)
+
+							x_origin = column * x_delta 
+							y_origin = row * y_delta
+		
+							delta_x_y = Base.Vector(x_origin, y_origin,0)
+		
+							polygon_copy = polygon.copy()
+							polygon_copy.translate(delta_x_y)		
+							fn=Part.Face([polygon_copy])
+							e_faces.append(fn)
+
+					else:
+					# even
+						if row <= (n_rows/2) and row >= (-n_rows/2):
+							centers_distance = tickness + 2 * radius * math.sin(edges_angle)
+							x_delta = centers_distance  * math.sin(edges_angle)
+							y_delta = centers_distance
+							
+							x_origin = column * x_delta 
+							y_origin = row * y_delta
+		
+							delta_x_y = Base.Vector(x_origin, y_origin,0)
+		
+							polygon_copy = polygon.copy()
+							polygon_copy.translate(delta_x_y)		
+							fn=Part.Face([polygon_copy])
+							e_faces.append(fn)
+
+		# join all the faces
+		mshell = Part.makeShell(e_faces)
+		extruded_polygon = mshell.extrude(Base.Vector(0,0,height))
+
+		# cut the array of solids using the container box
+		shape = container.cut(extruded_polygon) # comment it to see the array of solids
+
+		fp.Shape = shape # comment it to see the array of solids
+		#fp.Shape = extruded_polygon # uncomment it to see the array of solids
+
+	def __getstate__(self):
+		''' When saving the document this object gets stored using Python's cPickle module.
+		Since we have some un-pickable here -- the Coin stuff -- we must define this method
+		to return a tuple of all pickable objects or None.
+		'''
+		return self.Type, self.Version
+
+	def __setstate__(self,state):
+		''' When restoring the pickled object from document we have the chance to set some
+		internals here. Since no data were pickled nothing needs to be done here.
+		'''
+		self.Type = state[0]
+		self.Version = state[1]
+
+class ViewProviderHoneycombSolid:
+	def __init__(self, obj):
+		''' Set this object to the proxy object of the actual view provider '''
+		obj.Proxy = self
+
+	def attach(self, obj):
+		''' Setup the scene sub-graph of the view provider, this method is mandatory '''
+		return
+
+	def updateData(self, fp, prop):
+		''' If a property of the handled feature has changed we have the chance to handle this here '''
+		return
+
+	def getDisplayModes(self,obj):
+		''' Return a list of display modes. '''
+		modes=[]
+		return modes
+
+	def getDefaultDisplayMode(self):
+		''' Return the name of the default display mode. It must be defined in getDisplayModes. '''
+		return "Shaded"
+
+	def setDisplayMode(self,mode):
+		''' Map the display mode defined in attach with those defined in getDisplayModes.
+		Since they have the same names nothing needs to be done. This method is optional.
+		'''
+		return mode
+
+	def onChanged(self, vp, prop):
+		''' Print the name of the property that has changed '''
+		FreeCAD.Console.PrintMessage("Change property: " + str(prop) + "\n")
+
+	def getIcon(self):
```

Comment:
Please save this icon as a file along side the `HoneycombSolid.FCMacro` file (and add it to the list of files).

Reply:
Done!, and I created a new icon.